README.md

```diff
@@ -64,6 +64,116 @@ where:
 
 * namespace: The namespace knative-eventing resources will be deployed into (it cannot be deployed into the same namespace as knative-operator or knative-serving
 
+## Usage with the Gateway APIs
+
+Kubernetes, Istio, and other service meshes are working toward a combined [Gateway API](https://gateway-api.sigs.k8s.io/) for Kubernetes Ingress, Load Balancing, and Service Mesh.  This would replace Istio's Gateway, VirtualService, and other resources.  There are currently two ways to use these APIs with Knative, both described below. 
+
+### Using the new Gateway API directly
+
+Knative's ingress/routing is configured by [net extensions](https://github.com/knative-extensions?q=net&type=all&language=&sort=).  We can replace the default [net-istio](https://github.com/knative-extensions/net-istio) with [net-gateway-api](https://github.com/knative-extensions/net-gateway-api) and Knative will natively generate/use the Gateway API resources.  In this example, we use the `istio-k8s` and `istio-ingress-k8s` charms, which support the new Gateway API, to deploy Istio:
+
+Deploy istio:
+```bash
+juju add-model istio-system
+juju deploy istio-k8s -m istio-system --channel edge --trust
+juju deploy istio-ingress-k8s -m istio-system --channel edge --trust
+```
+
+Deploy Knative:
+```bash
+juju add-model knative-user
+juju deploy ./charms/knative-operator/knative-operator_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --resource knative-operator-image=gcr.io/knative-releases/knative.dev/operator/cmd/operator@sha256:18e7a0c612efdce11d9d396fa9a1469fa206bdb2a817b5426a595ed57a5c3daf \
+  --resource knative-operator-webhook-image=gcr.io/knative-releases/knative.dev/operator/cmd/webhook@sha256:dc2b23c9f66869a8617a51f2f0af0f56d9dc6a85b71cff587161eb9711aef26a
+juju deploy ./charms/knative-serving/knative-serving_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --config namespace=knative-serving-1 \
+  --config istio.gateway.namespace=istio-system \
+  --config domain.name=10.64.140.43.nip.io \  # <-- may be different if you have a different load balancer configuration
+  --config istio.gateway.name=istio-ingress-k8s \
```

Comment:
I'm assuming this means it won't really rely on relations with the ingress charm and directly spin `HTTPRoutes` that will connect to the gateway listener owned by the ingress charm, iirc when removing the ingress charm it won't really remove the routes it doesn't own (Knative's in this context), as they won't have the same labels, correct??

Reply:
yes, at present this charm will not rely on the ingress relation **to create `HTTPRoutes`**, although we should have some relation (either the existing ingress, or something else) that communicates just the gateway information.  That way we can do something like `juju relate knaitve:gateway-info istio-ingress:???` and then the config is automatically populated